pages/DSNP/Operations.md

```diff
@@ -0,0 +1,35 @@
+# Operations
+
+DSNP implementations perform well-defined DSNP Operations and generate DSNP State Change Records.
+
+## Control Keys and Proofs
+
+Each invocation of a DSNP Operation MUST have verifiable approval of the acting principal(s) via a Control Key Ownership Proof.
+The precise data type and representation of both the Control Key and the Control Key Ownership Proof MUST be defined by each DSNP implementation.
+For example, an implementation might use the public key of an asymmetric key pair as a control key, and provide a proof for each operation by producing a cryptographic signature of the user's DSNP Identifier and some nonce value.
+
+Where operations are listed as using control keys or ownership proofs as input parameters, this indicates that they should be provided in addition to those providing invocation authentication.
+
+## Transaction Identifiers
+
+Each invocation of a DSNP Operation should be associated with a transaction identifier. The precise data type and usage pattern of a transaction identifier is left implementation-dependent.
```

Comment:
I think having transaction identifier as implementation-dependent will cause conflicts and eventually problematic for inter-operability between different implementations of DSNP

Reply:
Syntax issue: Uppercase T and I on Transaction Identifier?

Reply:
> I think having transaction identifier as implementation-dependent will cause conflicts and eventually problematic for inter-operability between different implementations of DSNP

@aramikm I don't think it's much different from saying the control key algorithm is implementation-dependent. That said, if you think we can just say it will be a unique string, I'm fine with that. Let me know if I'm not understanding your concern here though.

Reply:
I think it would be good to have that uniqueness requirement.
An example of my concern:
We have two DSNP implementations of A and B. Both of them decide to use incremental u64 as a transaction identifier. While trying to Interoperate between each other we will have 2 different transactions for each ID.

I think one way to avoid this scenario is to prefix the implementation specific name into transaction identifier. For following example that might looks like 'A://1', 'B://1', 'A://2', 'B://2', ....

Reply:
> I think one way to avoid this scenario is to prefix the implementation specific name into transaction identifier. For mentioned example that might looks like 'A://1', 'B://1', 'A://2', 'B://2', ....

Could we use DID spec (or something like it, or something related) for that somehow?

Reply:
@aramikm Good idea... A prefix will give us a namespacing option for future interoperability. FWIW we also need this unique name to be incorporated into DSNP Content URLs to enable interoperability. The simplest thing to do in the context of this PR is to say that "dsnp://" is the prefix, because the current spec does not address interoperability. But I'd like to see interoperability (at least _potential_ interoperability) broken out into a separate DIP. This might entail Frequency changing to use something like "frqcy://" for its internal URLs/IRIs. To your point @shannonwells this implementation-specific prefix would need to be incorporated in the DID method. Options are "did:frqcy:123" or something like "did:dsnp:frqcy/123". The latter would in some respect require (or allow, depending on your point of view) a single "dsnp" DID method resolver that DSNP.org would maintain and allow the merging in of compliant implementations.

Reply:
Actually for now we can just ensure that Tx IDs are unique within an implementation, and then we can specify how they may be referenced from other implementations when we address interoperability (soon!).